app/src/main/java/pl/example/android/garbageapp/data/database/SectorTermsDatabaseDataSource.java

```diff
@@ -0,0 +1,80 @@
+package pl.example.android.garbageapp.data.database;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import pl.example.android.garbageapp.utils.AppExecutors;
+import pl.example.android.garbageapp.utils.NotificationUtils;
+import pl.example.android.garbageapp.utils.SectorTermsDateUtils;
+
+/**
+ * Created by miltomasz on 21/11/17.
+ */
+
+public class SectorTermsDatabaseDataSource {
+
+    private static final String LOG_TAG = SectorTermsDatabaseDataSource.class.getSimpleName();
+
+    private static final Object LOCK = new Object();
+    private static SectorTermsDatabaseDataSource sInstance;
+
+    private final AppExecutors mExecutors;
+    private final Context mContext;
+    private final SectorTermDao mSectorTermDao;
+
+    private SectorTermsDatabaseDataSource(Context context,
+                                          SectorTermDao sectorTermDao, AppExecutors executors) {
+        mContext = context;
+        mSectorTermDao = sectorTermDao;
+        mExecutors = executors;
+    }
+
+    public static SectorTermsDatabaseDataSource getInstance(Context context,
+                                                            SectorTermDao sectorTermDao,
+                                                            AppExecutors executors) {
+        Log.d(LOG_TAG, "Getting the database data source");
+        if (sInstance == null) {
+            synchronized (LOCK) {
+                sInstance = new SectorTermsDatabaseDataSource(
+                        context.getApplicationContext(), sectorTermDao, executors
+                );
+                Log.d(LOG_TAG, "Made new database data source");
+            }
+        }
+        return sInstance;
+    }
+
+    public void scheduleCountingSectorTermsForNotification() {
```

Comment:
Because of Alarm Manager disadvantage it should restarted after boot as shown below using broadcast receiver listening for android.intent.action.BOOT_COMPLETED
https://stackoverflow.com/questions/38365325/android-alarmmanager-alarms-after-reboot-2016

Reply:
Yes, I agree but as I mentioned in the headline comment I'd add restarting alarm manager after boot later on (not in this ticket).